lexer/impl/lexer.cpp

```diff
@@ -1,6 +1,309 @@
+                    break;
+                case ':':
+                    checkTwoCharToken(&index, word, ":=", TokenType::Column, TokenType::Assignment);
+                    break;
+
+                default:
+
+                    if (isdigit(c))
+                    { // if number constant
+                        index--;
+                        std::string num = readWhileNum(&index, word);
+                        TokenType type = TokenType::IntegerLiteral;
+                        if (index < int(word.length()) && word[index] == '.')
+                        { // if real
+                            int oldIndex = index;
+                            index++;
+                            std::string floatPart = readWhileNum(&index, word);
+                            if (floatPart.length() < 1)
+                            { //eg ..
+                                index = oldIndex;
+                                result.push_back(makeToken(type, lineNum, num));
+                                break;
+                            }
+                            num.append(".");
+                            num.append(floatPart);
+                            type = TokenType::RealLiteral;
+                        }
+                        result.push_back(makeToken(type, lineNum, num));
+                    }
+                    else if (iswalpha(c))
+                    { // if var name
+                        std::string name = toStr(c);
+                        while (index < int(word.length()) && (iswalpha(word[index]) || isdigit(word[index]) || word[index] == '_'))
+                        {
+                            name.append(toStr(word[index++]));
+                        }
+                        result.push_back(makeToken(TokenType::Identifier, lineNum, name));
+                    }
+                    else
+                    { // non-valid construction
+                        std::cout << "INVALID CHAR" << c << "\n";
+                        result.push_back(makeToken(TokenType::InvalidToken, lineNum, toStr(c)));
+                    }
+
+                    break;
+                }
+            }
+        }
+        return 0;
+    }
+
+    /**
+     * Processing a line of riddle code. It updates the 'result' vector of tokens
+     * 
+     * @param line, string with line
+    */
+    void processLine(std::string line)
+    {
+        lineNum++;
+        std::istringstream iss(line);
+        std::string word;
+        while (std::getline(iss, word, ' '))
+        {
+            if (processWord(word) == -1)
+            { //if get a single-line comment
+                return;
+            }
+        }
+        result.push_back(makeToken(TokenType::NewLine, lineNum, "\n"));
+    }
+} // namespace lexer
+
+/**
+ * Gets the trie from the lexer's namespace. 
+ * Method should be used for testing and debugging only
+ * 
+ * @return trie, used during parsing (debugging purposes)
+*/
+common::Trie<TokenType> getTrie()
```

Comment:
Why this function is out of `namespace lexer`. All library code related to lexer should be in that namespace.